internal/armada/configuration/types.go

```diff
@@ -124,10 +124,25 @@ type SchedulingConfig struct {
 	MaximumResourceFractionToSchedule map[string]float64
 	// Overrides MaximalClusterFractionToSchedule if set for the current pool.
 	MaximumResourceFractionToScheduleByPool map[string]map[string]float64
-	// Max number of jobs to schedule in each invocation of the scheduler.
-	MaximumJobsToSchedule uint
-	// Max number of gangs to schedule in each invocation of the scheduler.
-	MaximumGangsToSchedule uint
+	// Token bucket global job scheduling rate limiter settings; see
+	// https://pkg.go.dev/golang.org/x/time/rate#Limiter
+	//
+	// Rate-limiting is based on the number of tokens available at the start of each scheduling round,
+	// i.e., tokens accumulated while scheduling are only available at the start of the next scheduling round.
+	//
+	// MaximumSchedulingRate controls the number of jobs scheduled per second in steady-state,
+	// i.e., once the burst capacity has been exhausted.
+	MaximumSchedulingRate float64 `validate:"gt=0"`
```

Comment:
I understand why you've made the rate limits act over a given scheduling cycle, but I think longer term we want to make these apply over some timer horizon.  That's because what users and administrators care about isn't the scheduling rate in given scheduling cycle but rather the scheduling rate over various time horizons.

Reply:
I've updated the docstring to make clear the rate-limiter persists between rounds.